core/tests/puppeteer-acceptance-tests/user-utilities/exploration-editor-utils.ts

```diff
@@ -0,0 +1,331 @@
+const mobileStateNodeSelector = '.e2e-test-node-background';
+
+// Preview tab elements.
+const nextCardButton = '.e2e-test-next-card-button';
+const submitAnswerButton = '.e2e-test-submit-answer-button';
+const previewRestartButton = '.e2e-test-preview-restart-button';
+const stateConversationContent = '.e2e-test-conversation-content';
+const explorationCompletionToastMessage = '.e2e-test-lesson-completion-message';
+
+export class ExplorationEditor extends BaseUser {
+  /**
+   * Function to navigate to the creator dashboard page.
+   */
+  async navigateToCreatorDashboard(): Promise<void> {
+    await this.page.goto(creatorDashboardUrl);
+  }
+
+  /**
+   * Function to click on the 'Create Exploration' button.
+   */
+  async clickCreateExplorationButton(): Promise<void> {
+    await this.clickOn(createExplorationButton);
+    await this.clickOn(dismissWelcomeModalSelector);
+  }
+
+  /**
+   * Function to add content to a card.
+   * @param {string} text - The content to be added to the card.
+   */
+  async updateCardContent(text: string): Promise<void> {
+    await this.clickOn(stateEditSelector);
+    await this.waitForElementToBeClickable(stateContentInputField);
+    await this.type(stateContentInputField, `${text}`);
+    await this.clickOn(saveContentButton);
+    await this.page.waitForSelector(stateContentInputField, {hidden: true});
+  }
+
+  /**
+   * Function to add an interaction to the exploration.
+   * @param {string} interactionToAdd - The interaction type to add to the Exploration.
+   * Note: A space is added before and after the interaction name to match the format in the UI.
+   */
+  async addInteraction(interactionToAdd: string): Promise<void> {
+    await this.clickOn(addInteractionButton);
+    await this.clickOn(` ${interactionToAdd} `);
+    await this.clickOn(saveInteractionButton);
+    await this.page.waitForSelector(saveInteractionButton, {hidden: true});
+  }
+
+  /**
+   * Function to display the Oppia responses section.
+   */
+  async viewOppiaResponses(): Promise<void> {
+    await this.clickOn(stateResponsesSelector);
+  }
+
+  /**
+   * Function to select the card that learners will be directed to from the current card.
+   * @param {string} cardName - The name of the card to which learners will be directed.
+   */
+  async directLearnersToNewCard(cardName: string): Promise<void> {
+    await this.clickOn(openOutcomeDestButton);
+    await this.waitForElementToBeClickable(destinationCardSelector);
+    // The '/' value is used to select the 'a new card called' option in the dropdown.
+    await this.select(destinationCardSelector, '/');
+    await this.type(addStateInput, cardName);
+    await this.clickOn(saveOutcomeDestButton);
+  }
+
+  /**
+   * Function to save an exploration draft.
+   */
+  async saveExplorationDraft(): Promise<void> {
+    if (this.isViewportAtMobileWidth()) {
+      const element = await this.page.$(mobileNavbarOptions);
+      if (!element) {
```

Comment:
In what circumstance is there no element present? Maybe explain that in a comment.

Reply:
Done.